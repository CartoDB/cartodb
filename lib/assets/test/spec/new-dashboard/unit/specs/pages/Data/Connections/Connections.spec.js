import Vuex from 'vuex';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import ConnectionsComponent from 'new-dashboard/pages/Data/Connections/Connections';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Connections.vue', () => {
  beforeEach(() => {
  });

  describe('Render', () => {
    it('should render initial state', () => {
      const store = createStore({
        connectors: {
          loadingConnections: false,
          connections: []
        },
        config: {
          isFirstTimeViewingDashboard: false
        }
      });
      const connectionsList = createConnections(store);

      expect(connectionsList).toMatchSnapshot();
    });

    it('should render one elements', () => {
      const store = createStore({
        connectors: {
          loadingConnections: false,
          connections: [{
            id: 'IDENTIFIER',
            name: 'CONNECTION NAME',
            connector: 'mysql',
            type: 'db-connector',
            parameters: {
              billing_project: 'BILLING_PROJECT',
              default_project: 'DEFAULT_PROJECT',
              service_account: '********'
            }
          }]
        },
        config: {
          isFirstTimeViewingDashboard: false
        }
      });
      const connectionsList = createConnections(store);

      expect(connectionsList).toMatchSnapshot();
    });
  });

  describe('Navigation', () => {
    it('should navigate to edit page', () => {
      const store = createStore({
        connectors: {
          loadingConnections: false,
          connections: [{
            id: 'IDENTIFIER',
            name: 'CONNECTION NAME',
            connector: 'bigquery',
            type: 'db-connector',
            parameters: {
              billing_project: 'BILLING_PROJECT',
              default_project: 'DEFAULT_PROJECT',
              service_account: '********'
            }
          }]
        },
        config: {
          isFirstTimeViewingDashboard: false
        }
      });
      const connectionsList = createConnections(store);
      connectionsList.find('#IDENTIFIER').trigger('click');

      expect(connectionsList.vm.$router.push).toHaveBeenCalledWith({ name: 'edit-connection', params: { id: 'IDENTIFIER' } });
    });

    it('not should navigate when connection is using OAUTH', () => {
      const store = createStore({
        connectors: {
          loadingConnections: false,
          connections: [{
            id: 'IDENTIFIER',
            name: 'CONNECTION NAME',
            connector: 'gdrive',
            type: 'db-connector',
            parameters: {
              billing_project: 'BILLING_PROJECT',
              default_project: 'DEFAULT_PROJECT',
              service_account: '********'
            }
          }]
        },
        config: {
          isFirstTimeViewingDashboard: false
        }
      });
      const connectionsList = createConnections(store);
      connectionsList.find('#IDENTIFIER').trigger('click');

      expect(connectionsList.vm.$router.push).not.toHaveBeenCalled();
    });
  });
});

function createStore (customStoreData) {
  const storeData = {
    connectors: {
      connections: null,
      projects: null,
      bqDatasets: null,
      loadingConnections: false,
      loadingProjects: true
    },
    config: {
      isFirstTimeViewingDashboard: false
    },
    user: {
      account_type: 'account_type'
    }
  };

  const storeInstance = new Vuex.Store({
    state: customStoreData || storeData
  });

  return storeInstance;
}

function createConnections (store, propsData) {
  return shallowMount(ConnectionsComponent, {
    mocks: {
      $t: key => key,
      $router: {
        push: jest.fn()
      }
    },
    store,
    localVue,
    propsData
  });
}
