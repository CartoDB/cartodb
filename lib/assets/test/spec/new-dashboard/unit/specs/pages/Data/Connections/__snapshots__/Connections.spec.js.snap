// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connections.vue Render should render initial state 1`] = `
<section class="u-mt--64">
  <div class="container grid">
    <div class="u-width--100">
      <sectiontitle-stub class="grid-cell"><template></template> <template></template> <template></template></sectiontitle-stub>
    </div>
    <div class="u-width--100 u-mt-50 grid-cell">
      <div class="emptyState u-pl--64 u-mb--60">
        <h3 class="is-body is-semibold u-mt--64">DataPage.startAddingConnections</h3>
        <p class="u-mt--16 is-caption">DataPage.connectDescription</p>
        <router-link to="[object Object]"><button class="button is-primary u-mt--48" style="margin-bottom: 56px;"> DataPage.newConnection </button></router-link> <img width="314" src="../../../assets/images/connectors/connectors.png" class="logo">
      </div>
    </div>
  </div>
  <router-view></router-view>
</section>
`;

exports[`Connections.vue Render should render one elements 1`] = `
<section class="u-mt--64">
  <div class="container grid">
    <div class="u-width--100">
      <sectiontitle-stub class="grid-cell"><template></template> <template></template> <template></template></sectiontitle-stub>
    </div>
    <div class="u-width--100 u-mt-50 grid-cell">
      <div class="connections-container grid">
        <div class="connector-wrapper grid-cell grid-cell--col4">
          <connection-stub id="IDENTIFIER" type="mysql" connectiontype="database" connectionname="CONNECTION NAME" connectionparams="[object Object]" label="MySQL"></connection-stub>
        </div>
      </div>
    </div>
  </div>
  <router-view></router-view>
</section>
`;
